pkg/labels/consul_aggregator.go

```diff
@@ -111,6 +151,11 @@ func (c *consulAggregator) removeWatch(watch *selectorWatch) {
 	} else {
 		c.watchers.delete(watch)
 	}
+	if c.watchers != nil {
+		c.metWatchCount.Update(int64(c.watchers.len()))
+	} else {
+		c.metWatchCount.Update(0)
```

Comment:
they way you've written `len` it may actually have been safe to call `len` on nil, but I'll not complain about this since this way is sure to be safe